modules/transactionpool/persist.go

```diff
@@ -53,6 +55,9 @@ func (tp *TransactionPool) initPersist() error {
 	}
 
 	// Open the database file.
+	if build.DEBUG && build.Release != "testing" {
+		fmt.Println("\tTransaction Pool: opening database")
+	}
```

Comment:
I don't think this is the best way to debug the problem. Println debugging is not effective against hard-to-reproduce bugs. I suggest running with profiling enabled so that you can inspect the blocking profile live when it hangs. That should tell you exactly where it got stuck.